Ansible core

jturmelle opened the pull request:
Documentation updates for the centos9/ansible11 updates found in the [PR](https://github.com/iridl/iridl-ansible/pull/18).

This has significant changes including documenting how to Integrate Bitbucket keys.

jturmelle committed:
changes for the new Ansible implementation under CentOS Stream 9

jturmelle committed:
Add firewall commands

jturmelle committed:
updated copyright

jturmelle committed:
updated for CentOS9

jturmelle committed:
updated for CentOS9

jturmelle committed:
updated to include sphinx-copybutton and bring python packages up to date.

jturmelle committed:
updated to include sphinx-copybutton and bring python packages up to date. Added extra copybutton files I missed.

jturmelle committed:
Make sure they use the correct virtuall environment when updating.

aaron-kaplan commented on docsrc/admin/git.md:
```diff
@@ -0,0 +1,52 @@
+
+The person installing and maintaining the Data Library should have or [create an account](https://id.atlassian.com/signup?application=bitbucket&continue=https%3A%2F%2Fbitbucket.org%2Faccount%2Fsignin%2F%3Fnext%3D%252F%26redirectCount%3D1) with [https://bitbucket.org](https://bitbucket.org).
+
+Once you have an account inform the [Data Library staff](mailto:help@iri.coplumbia.edu) to grant you access to the repositories.
```
`columbia` is misspelled.

aaron-kaplan commented on docsrc/admin/git.md:
```diff
@@ -0,0 +1,52 @@
+      ssh-keygen -t ed25519 -b 4096 -f ~/.ssh/id_bitbucket
+
+  _This will create an ssh key for use with bitbucket in ~/.ssh/id_bitbucket. Set a good passphrase different
+  from your password. You will need to use this passphrase every time you need to commit changes to the bitbucket repository,
```
Is it really important that it be different from the password? Fewer passwords to remember seems better.

The point will be moot if we switch to the OAuth mechanism, which should be possible here since these docs are now for CentOS 9. Please give that a try.

jturmelle replied:
The reason I did this was in case they're using a shared account to deploy changes.  I didn't want someone to share their personal  password here. What should our position be on this?  Should they create one account that everyone shares to deploy changes, or if there are multiple people who can deploy changes, should they each have their own instance?  There are pros and cons to each method.  I will try the oauth mechanism, which implies that each member has their own instance.

aaron-kaplan replied:
I have tolerated a single shared account when the partner actively preferred that, but I don't want to encourage that approach. Let's assume separate accounts and just make sure we emphasize that you have to git pull before running the playbook, and commit and push when you make changes.

aaron-kaplan commented on docsrc/admin/installation.md:
```diff
@@ -2,85 +2,83 @@
+      sudo setenforce permissive
 
-* Disable SELinux and reboot:
+  **Note:** If you feel SELinux is important to the security of your server, please start a conversation with us at help@iri.columbia.edu.
```
I have yet to encounter anyone at a national met service who knows what SELinux is, let alone perfers to leave it enabled. Maybe we can remove this note.

jturmelle replied:
That sounds preferable because we don't want to really have that conversation!

aaron-kaplan commented on docsrc/admin/installation.md:
```diff
@@ -2,85 +2,83 @@
+      sudo dnf install -y git python3.12
+      sudo python3.12 -m venv /opt/datalib_venv3.12
+      sudo /opt/datalib_venv3.12/bin/pip install --upgrade pip
+      sudo /opt/datalib_venv3.12/bin/pip install ansible==11.3.0 requests==2.32.3
```
I thought we were going to use the CentOS packaging of ansible.

aaron-kaplan commented on docsrc/admin/installation.md:
```diff
@@ -90,45 +88,35 @@ If you feel SELinux is important to the security of your server, please start a
 
-* Commit your customizations and push them to your git server for safe keeping;
-  back up `secrets.yaml` by other means,
+* Commit your customizations and push them to your git server for safe keeping; back up `secrets.yaml` by other means, 
```
I feel bad when I read this. It essentially says "We don't know how to manage secrets. You're on your own."

jturmelle replied:
If we do as you suggested: create a script/alias to run the ansible-playbook command, we could embed an encryption/decryption command for the secrets.yaml file.  Similar to what we do with your apb script.

aaron-kaplan replied:
And introduce yet another password for them to forget, or confuse with their server password and their ssh key encryption password? And if there are multiple admins, how do they share the secrets? I can't recommend git-secret because it's hard to use, but it solves a real problem.

aaron-kaplan commented on docsrc/admin/overview.md:
```diff
@@ -1,22 +1,21 @@
+the software on other platforms.
+
+_CentOS7 is beyond its end of life and is no longer recommended to install, but we continue to support the 
+installations where it is already installed. Any new installations must be made with CentOS Stream 9_
```
We don't commit to supporting anyone with whom we don't have a funded project. We need CentOS 7 --> 9 upgrade instructions, and a link to them here.

jturmelle replied:
I don't know if we can have generic upgrade instructions, but I'll give it a try.

aaron-kaplan commented on docsrc/admin/overview.md:
```diff
@@ -1,22 +1,21 @@
-The Data Library services runs under Docker, using docker compose. Most services
-log to stdout, which the Docker daemon
+The Data Library services runs under Docker, using docker compose. Most services log to stdout, which the Docker daemon
 forwards to journald.
```
Note the mention of journald. Need to make it true, or failing that, change this doc.

aaron-kaplan commented on docsrc/conf.py:
```diff
@@ -31,6 +31,7 @@
     "myst_parser",
     "sphinxcontrib.fulltoc",
     "sphinx.ext.githubpages",
+    "sphinx_copybutton",
```
Nice.

aaron-kaplan commented on docsrc/admin/git.md:
```diff
@@ -0,0 +1,52 @@
+# Configuring Git
```
There are also git configuration instructions in the install guide. That's for configuring deploy keys for use by the update script, this is for configuring personal keys for use by the administrator. Might need to explain that.

jturmelle replied:
Agreed. I actually got a little confused when I was running through the instructions.

jturmelle committed:
Significant changes including configuration changes to change the theme to the "Read the Docs" theme, added google analytics, autolabeling of sections and IRI branding.  Added Notes and SeeAlso sections to improve readbility.

Modified the structure to move server installation outside the DL Installation and more extensive instructions on git and deploy keys.

Included instructions for the new install.bash command.

I've begun instructions for upgrading, but they are not included yet too avoid confusion.

aaron-kaplan commented:
I think the ssh key management stuff is confusing, but I'll wait to comment on that until you've experimented with the oauth workflow and decided whether to switch to that. BTW if we use the oauth workflow, we probably still need to create the deploy key, but we may be able to set it up programmatically using oauth and a bitbucket web API.

Moving the server specs and OS installation instructions to the end seems like a weird move, since you have to buy the server and install the OS before you can do anything else.

jturmelle committed:
updated to use additional extensions and RTD theme.

jturmelle committed:
changed install.bash to run_ansible

jturmelle committed:
source the python virtual environment before running ansible-galaxy

jturmelle committed:
Reformatted files to 120 columns. Started some upgrade instructions, but have not referenced them. Have not attempted git oauth yet.

jturmelle committed:
Reformatted files to 120 columns. Started some upgrade instructions, but have not referenced them. Have not attempted git oauth yet.

jturmelle commented:
Not ready for merging yet, just saving my work.  I still need to test and document OAuth functionality. 

Regarding the sidebar content that you've commented on, I've placed the Server Installation at the end because it's the least used.  Once they've performed the Server Install, they never need to look at it again.  I was tempted to not even put it on the main menu, but right now just about everyone needs to do the Server Installation, so it's prominent, but not ordered. I've also reformatted all the text to 120 columns, I'm not sure if that works for you or not.

I included an extension for autosectionlabel, so I've removed a lot of the "label" code, and I've tried to use {ref} and {doc} tags everywhere to reference internal links.

There's a major issue where flags with two dashes look like one dash.  I've tried manipulating fonts, but haven't been able to fix it yet. 

Still working on the upgrade notes.  I have some copy, but it's not referenced in the site yet.

aaron-kaplan commented:
@jturmelle I think this is the PR that was mentioned on the IMD email thread. We can move ahead without 7-to-9 upgrade notes, but your last comment on this PR says "not ready for merging yet, just saving my work," so please finish whatever else there is to finish.

I would strongly prefer to have the chapters in chronological order rather than order of frequency of use.

jturmelle committed:
remove the upgrade help.  This is not really helpful and too complicated

jturmelle committed:
remove upgrade help as its toocomplicated

jturmelle committed:
remove upgrades as its too complicated

jturmelle committed:
Data Library updates for CentOS Stream, including instructions for upgrading from CentOS7

jturmelle committed:
modifications for CentOS Stream9/10 Installation

jturmelle committed:
updated packages to address dependabot alerts from github

aaron-kaplan commented:
@jturmelle is this ready for another round of review or are you still working on it?

jturmelle commented:
Yes, I think it's ready for review.

jturmelle commented:
Accidentally hit the close pull request.  I meant to just say it is ready for review.  I've been looking at it so much that I'm sure there are issues but I can't see them any more.

aaron-kaplan commented:
OK if I push some suggested changes to this branch? I think that would be more efficient than asking you to make them via comments.

jturmelle commented:
> OK if I push some suggested changes to this branch? I think that would be more efficient than asking you to make them via comments.

Absolutely, please do.

aaron-kaplan commented on docsrc/admin/git.md:
```diff
@@ -0,0 +1,108 @@
+1. bitbucket_private_key for the dlentries and classic maprooms.
+2. python_maproom_private_key for the python maprooms.
+
+To create a personal deployment key:
```
Either this was a typo or there's an important misunderstanding here. Deployment keys are *not* personal. They're associated with a repo, not with a person.

jturmelle replied:
It was a misunderstanding, I've updated the text.

aaron-kaplan commented on docsrc/admin/git.md:
```diff
@@ -0,0 +1,108 @@
+
+Leave the passphrase blank.
+
+### Add the Access or Deploy Key to Bitbucket or Github
```
Let's not complicate things by talking about Github. We only used Github once, with Lesotho. I don't think it makes sense to complicate everyone else's documentation for the sake of Lesotho.

jturmelle replied:
Done.

aaron-kaplan commented on docsrc/admin/git.md:
```diff
@@ -0,0 +1,108 @@
+
+### Add the Access or Deploy Key to Bitbucket or Github
+
+In the following, the _xxx_ is your repository code. For example: _kmd_, _nimet_, _madagascar_, etc.
```
backticks instead  of underscores?

jturmelle replied:
Either way.  I just wanted to highlight them, so I italicized them.

jturmelle replied:
I've reformatted all content to use backticks where emphasis or highlighting was desired.

aaron-kaplan commented on docsrc/admin/installation.md:
````diff
@@ -1,165 +1,156 @@
+
+  ```{seealso}
+  The secrets.yaml file contains the deploy keys (or access keys) to access the repositories defined in your playbook.yaml
+  file.  These are needed because Ansible runs as root, so it needs it's own access to the repositories.  For that reason,
````
it's --> its

I'm not sure it's necessary to explain why deploy keys are needed, but if we are going to explain it, the fact that ansible runs as root doesn't really have anything to do with it. Deploy keys are needed because `update_datalib` needs permission to pull, and it is intended to be run by users who don't necessarily have their private keys on the server.

jturmelle replied:
I was trying to make the same point, that root doesn't have permission to pull.  Because that happens to be the user that will be running update_datalib.

jturmelle replied:
agreed, removed deploy keys description from here.

aaron-kaplan commented on docsrc/admin/installation.md:
```diff
@@ -1,165 +1,156 @@
+  * Don't upgrade without checking the release notes first, because in some
+  cases an upgrade may require manual migration steps. (At this writing, 
+  there are no upgrade release notes because this is the playbook's initial 
+  release.)
```
We never wrote any release notes, or a versioned release for that matter. Maybe it's time to stop pretending that one day we will.

jturmelle replied:
You never know.  There may be caveats we want to highlight before people upgrade.

jturmelle replied:
Removed.

aaron-kaplan commented on docsrc/admin/installation.md:
````diff
@@ -1,165 +1,156 @@
-command:
+command
+```
+./run_ansible --build
````
`run-ansible` with a hyphen, here and elsewhere.

aaron-kaplan commented on docsrc/admin/installation.md:
```diff
@@ -1,165 +1,156 @@
+
+**For example**
+
+If there are errors or failures, you can pass *-vvv* to the run-ansible 
```
backticks instead of asterisks

jturmelle replied:
Don't backticks mean <code> formatting?  In this case, I'm highlighting in bold.

aaron-kaplan replied:
Backticks indicate code. It's usually expressed as a monospaced font, and sometimes grey highlighting. We currently indicate code using bold in some places, italics in others, and monospace font in yet others. I'm asking that we standardize on the backticks.

Single backticks are for in-line code, triple backticks are for code blocks.

aaron-kaplan commented on docsrc/admin/installation_upgrade.md:
```diff
@@ -0,0 +1,96 @@
+This section provides instructions for upgrading Data Library software previously installed on a 
+CentOS Stream 7 server to a server configured with CentOS Stream 9 or 10.  Make sure your current 
+dlconfig repository is pushed to the server and your secrets.yaml file is saved.  It is important to 
+back up the entire account as well as all your data files and catalogs  (typically in the /data directory)
```
Data catalogs don't need to be backed up, as they're kept in git (dlentries). Or maybe I didn't understand what you meant by "catalogs."

It's not just "typically" in /data, it's always in /data.

/data should always be a separate volume, so while it's never a bad idea to back it up, it should be possible to install a new OS on the root partition without wiping out /data.

In retrospect we should have had them put /home on a separate volume, but in most places we didn't, so they'll most likely want to back up and restore that.

I seem to remember dumping and un-dumping the database in past upgrades, but I think when upgrading from 7 to 9/10 that should no longer be necessary, because now we put postgres' storage on /data?

jturmelle replied:
Are you sure it's always in /data?  I feel like I've seen /local/data before.  There are some very old installations out there, and we can't assume that everything has been installed the same way. 

While it should be possible to install an OS on the root partition without wiping out /data, the installation process does reformat partitions, so I would never suggest to not to back it up before re-installing. We should definitely not imply in any way they don't need to backup first.

I agree, dumping the DB is a good idea and should be part of the process.  Even if we configure Postgres storage on /data, there's a good chance it could get wiped out in the re-install process.

aaron-kaplan replied:
If this is a guide on upgrading from ansibilized CentOS 7, then it's always /data. On the manually-configured CentOS 6 systems it was usually (perhaps always) /local, but that's a different upgrade procedure anyway.

aaron-kaplan commented on docsrc/admin/overview.md:
```diff
@@ -1,29 +1,65 @@
+installations must be made with CentOS Stream 9 or CentOS Stream 10. Please read the documentation
+for each Operating System to see which is more appropriate for you.  
+* CentOS Stream 10 will not work on older architectures (AMD/Intel 64-bit x86_64_v2 or below).
+* CentOS Stream 9 end-of-life is May 31, 2027_
```
Is the trailing underscore intentional? If so, what does it do?

jturmelle replied:
unintentional.

aaron-kaplan commented on docsrc/server/centos_stream.md:
```diff
@@ -0,0 +1,64 @@
+(install_centos9)=
+
+# Install CentOS Stream 9 or CentOS Stream 10
```
Let's make the docs say 10 everywhere, and just add a note that says if your hardware isn't supported by 10 you can also use 9.

aaron-kaplan commented on docsrc/server/centos_stream.md:
```diff
@@ -0,0 +1,64 @@
+
+#### Reboot
+
+  If any changes were installed during the package updates, reboot now.
```
Doesn't matter if any updates were installed--we have to reboot for the selinux change to take effect.

jturmelle replied:
You don't have to reboot immediately any more.  I use `setenforce` in one of the playbooks so selinux is not enforced after the playbook is run.

aaron-kaplan replied:
`setenforce` without a reboot only puts it in `permissive` mode, in which policies are loaded and violations are logged, but policy isn't enforced. The only way to get it into `disabled` mode is to reboot.

I'm not aware of any difference between permissive and disabled that would affect our applications, but if there ever is one, I want to discover it immediately, not months later when the system reboots for the first time.

jturmelle replied:
In fact, the instructions currently set selinux to permissive mode.  We never set it to disabled. 
server/centos_stream.html#disable-selinux

aaron-kaplan replied:
Oh ok, didn't notice that. We use "disabled" at IRI. I know because every time I run ansible it complains that one of the hosts needs to be rebooted for the selinux change to take effect :-/

aaron-kaplan commented on docsrc/admin/maintenance.md:
```diff
@@ -4,62 +4,69 @@
 
-        cd dlconfig
-        git pull --ff-only
-  
```
Just curious, why did you rewrite all the code blocks with backticks?

jturmelle replied:
I don't remember, but if I changed them all then I did it for consistency. Since I was going through every page to fix links and such, I guess I just made them all the way I was doing it.

jturmelle replied:
I remember now.  I wanted to preserve indentation in certain places which required using the backticks.  So then I started to do it for consistency.  All code blocks now use backticks.

aaron-kaplan commented on docsrc/admin/maintenance.md:
````diff
@@ -4,62 +4,69 @@
 of your dlconfig repository:
+```
+ansible-galaxy collection install iridl.iridl:==x.y.z
+```
````
Same note as above regarding release notes, and versioning. We should just give instructions for updating to the latest.

aaron-kaplan commented on docsrc/admin/maintenance.md:
```diff
@@ -70,45 +77,48 @@ personal data catalog directory (see {ref}`paths`) for the user. Ansible does
   If the table doesn’t exist, either run the sql script that creates it, or add
   it to the sql scripts if it’s missing. If the table exists, check that the
-  readonlyaccess role has select permission for it. (Our install process is
+  **readonlyaccess** role has select permission for it. (Our install process is
```
backticks?

aaron-kaplan reviewed (commented):
@jturmelle I will make the suggested changes myself, but I've added comments in case you want to discuss any of them. Please send whatever comments you can on Friday, because I'm going to publish a new version so I can reference it during the training next week.

jturmelle commented:
I think you probably have better t things to do than implement these fixes.  Unless you've already done it, I can fix them now.

aaron-kaplan commented:
If you have time, I'd appreciate it. Please push whatever you have at the end of the day, as I'll probably work on it while travelling tomorrow.

aaron-kaplan commented:
Please prioritize reviewing https://github.com/iridl/iridl-ansible/pull/29 over addressing comments on this PR.

jturmelle commented:
> Please prioritize reviewing [iridl/iridl-ansible#29](https://github.com/iridl/iridl-ansible/pull/29) over addressing comments on this PR.

Thanks, that must have got lost in my Inbox.  I'm looking at it now.

jturmelle committed:
Modifications to address comments, standardize syntax to use <code> formatting exclusively for highlighting and emphasizing text, use the triple-backtick syntax for multiline code blocks, and use * for bullet points.
Remove most references to CentOS9.

jturmelle committed:
Add some notes for backup.  I don't know how to do the database backup.  I'll have to look closer at this.

aaron-kaplan commented:
Regarding backing up the db, I think backing up /data is sufficient. It's almost entirely a read-only database, so I don't think there's much risk of getting an inconsistent snapshot that way, but if you want to be 100% sure then just shut down postgres before taking the backup.

aaron-kaplan commented:
OMG you replaced all the italics and bold with backticks?! That is not what I meant! Backticks are for code, things that are meant to be typed in verbatim.

jturmelle commented:
It seemed wrong to me when I was doing it, but I completely misunderstood your comments regarding bacticks.  I’ll fix them, so just ignore that for now.J.On Sep 27, 2025, at 4:06 PM, Aaron Kaplan ***@***.***> wrote:﻿aaron-kaplan left a comment (iridl/iridl.github.io#14)
OMG you replaced all the italics and bold with backticks?! That is not what I meant! Backticks are for code, things that are meant to be typed in verbatim.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

aaron-kaplan commented:
I’ve started an editing pass, so just let me fix them, otherwise we’ll
probably have conflicts.

On Sat, Sep 27, 2025 at 4:39 PM Jeff Turmelle ***@***.***>
wrote:

> *jturmelle* left a comment (iridl/iridl.github.io#14)
> <https://github.com/iridl/iridl.github.io/pull/14#issuecomment-3342031927>
> It seemed wrong to me when I was doing it, but I completely misunderstood
> your comments regarding bacticks.  I’ll fix them, so just ignore that for
> now.J.On Sep 27, 2025, at 4:06 PM, Aaron Kaplan ***@***.***>
> wrote:﻿aaron-kaplan left a comment (iridl/iridl.github.io#14)
> OMG you replaced all the italics and bold with backticks?! That is not
> what I meant! Backticks are for code, things that are meant to be typed in
> verbatim.
>
> —Reply to this email directly, view it on GitHub, or unsubscribe.You are
> receiving this because you were mentioned.Message ID: ***@***.***>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/iridl/iridl.github.io/pull/14#issuecomment-3342031927>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAF3DRT4TLUR5VKMPQZ3LBL3U3YX7AVCNFSM6AAAAABXZGYYJCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTGNBSGAZTCOJSG4>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

jturmelle commented:
Ok sorry about that.

aaron-kaplan commented on docs/_sources/admin/maintenance.md.txt:
````diff
@@ -4,62 +4,69 @@
+this line to the end of your ~/.bash_profile file.
+```
+source /opt/datalib_venv3.12/bin/activate
+```
````
Is that safe? I'm a little concerned that activating a venv by default could break system utils that are python-based. But I don't have anything specific in mind.

jturmelle replied:
It's safe, but it may also cause problems if they're not familiar with virtual environments and want to use this account for python development. This is only per-user and won't affect anything in the system.   But you're probably right, it's probably best not to do this here.  We can tell them to activate the environment, and add a comment somewhere that if the ansible-playbook (or ansible-galaxy) command isn't working, make sure they've activated the environment.

aaron-kaplan committed:
Did a full copy-editing pass through the whole document.

aaron-kaplan committed:
Rebuild

aaron-kaplan commented:
I've finished a pass through the whole thing.

I still find the top-level structure confusing, so I'm going to play around with that, but I'll make those changes in a separate commit so you can see what I've actually changed vs. what I just moved around.

aaron-kaplan commented on docsrc/admin/installation_upgrade.md:
````diff
@@ -0,0 +1,104 @@
+
+  ```
+  git clone git@bitbucket.org:iridl/dlconfig_myconfig.git dlconfig
+  cd dlconfig
````
If /home was preserved through the OS upgrade or restored from backup after it, then it shouldn't be necessary to redo the git configuration or reclone the dlconfig repo. I think this should just say to do a git pull to make sure you're up to date. Disagree?

aaron-kaplan replied:
Also we should recommend putting /home on a separate volume so that the OS can be upgraded without wiping /home? But then we have to decide how big /home should be.

jturmelle replied:
I agree about using git pull after restoring their home directories.

I don't agree with trying to maintain the /home partition through the OS upgrade.  It's too easy to lose your partition and too difficult to explain to someone how to partition the disk properly while recovering LVM partitions.  No matter what, I need to explain the backup/restore process.

I was planning on doing a full walkthrough, with images, of an OS Install/upgrade.  Disk partitioning is the hardest since we don't know their disk size, whether they have multiple disks or the amount of data they have or intend to have.  Trying to cover all the bases regarding disk partitioning is too complicated.  I intended to show them one way (with an option for a RAID1) to do the install/upgrade.  If they are experienced enough, they can decide the best options for them for partitioning.

aaron-kaplan replied:
OK, I won't say anything about preserving /home.

I don't think we need a full walkthrough of the OS installation--other than configuring storage, I don't recall anyone having trouble with that. I agree that more guidance on how to configure storage could be useful.

aaron-kaplan committed:
reorganize

aaron-kaplan committed:
rebuild

aaron-kaplan committed:
gitignore

aaron-kaplan commented on docsrc/admin/installation_upgrade.md:
````diff
@@ -0,0 +1,104 @@
+  repository.  The `-b` flag will make a backup of your current files:
+
+  ```
+  cp -b ansible_collections/iridl/iridl/example/* .
````
Instead of copying with `cp -b` and then restoring some of the files from the backups, I think it would be less error-prone to just copy the specific files that need copying. I think that's `ansible.cfg`, `inventory.cfg`, and `run-ansible`. OK?

jturmelle replied:
That's fine. I was just trying to cover all the bases in case this changes.

aaron-kaplan committed:
More edits

aaron-kaplan committed:
rebuild

aaron-kaplan commented:
@jturmelle in one commit I did my big reorg with no content changes except for fixing references that I broke; then in another commit I made some more content edits. Please review.

jturmelle commented on docsrc/admin/installation_upgrade.md:
```diff
@@ -0,0 +1,91 @@
+
+* `/home`
+* `/data`
+* a dump of the postgres database
```
I think we agreed to move this line as we don't have a way to backup the DB as of now.

aaron-kaplan replied:
The last exchange I remember having about it was this one

https://github.com/iridl/iridl.github.io/pull/14#discussion_r2382786290

in which you said it would be better to dump it. I was about to ask you to fill in the details. If you think it's ok to just let postgres continue where it left off, under the assumption that /data was either preserved or restored from backup, then that's ok with me. I'll remove it for now, and if you think we should add instructions for dumping and undumping then we can do that in a follow-up PR.

jturmelle replied:
Let's follow up in a PR as I need to write a script that will dump/restore the DB.  It's something we should be able to do anyway.

jturmelle commented on docsrc/user/index.md:
```diff
@@ -1,58 +1,90 @@
 
 ## Dataset and maproom development
 
 ## Catalog entries
```
Should be a level 3 heading.

aaron-kaplan replied:
fixed

jturmelle commented on docsrc/user/index.md:
```diff
@@ -1,58 +1,90 @@
+in a browser, in order to validate your changes before pushing them to your git host. Your personal data catalog can
+reference files in either `/data/datalib/data` or `/data/datalib/home/<username>`.
 
 ## Maprooms
```
Should be a level 3 heading.

jturmelle commented on docsrc/user/index.md:
```diff
@@ -1,58 +1,90 @@
 
 (update-script)=
+
 ## Installing updates
```
Should be a level 3 heading.

jturmelle committed:
Modified the instructions to create a bootable flash drive.

jturmelle reviewed (commented):
I made a change in admin/centos_stream for creating a bootable disk as the old link was gone.  I added a couple comments about header levels, in the User Guide.  If you agree that those headers should be increased, let me know and I'll make those changes and build.

The only other changes I would make are in the disk partitioning.  But I think this should wait for another PR as I think I want to add pictures and increase the detail.

Besides that, I think everything looks good.

aaron-kaplan committed:
Don't dump and undump the DB

aaron-kaplan committed:
Fix heading levels in user guide

aaron-kaplan committed:
Copy changes in user guide

aaron-kaplan committed:
Rebuild

aaron-kaplan commented:
Made the changes we discussed in previous comments, plus a couple edits in the user guide to reflect changes we made a while ago: https://github.com/iridl/iridl.github.io/pull/14/commits/91f7b6d0ff6821c89959f36b3beb7ad017b5c12e
